mastermind.rb

```diff
@@ -0,0 +1,133 @@
+# Command Line Mastermind Game
+class Mastermind
+  attr_reader :player_score, :master_score, :player_guess,
+              :game_count, :board, :guess_count, :guess
+
+  def initialize
+    set_master_code
+    reset_board
+    reset_scores
+    reset_guess_count
+    display_rules
+    start_game
+  end
+
+  def reset_guess_count
+    @guess_count = 0
+  end
+
+  def set_master_code
+    @master_code = Array.new(4) { rand(2...8) }
+  end
+
+  def reset_scores
+    @player_score = 0
+    @master_score = 0
+  end
+
+  def reset_board
+    @board = Array.new(12) { Array.new(4) }
+    @win = false
+  end
+
+  def display_rules
+    print 'Welcome to Mastermind! A 4 letter code has been set by the mastermind. ' \
+          'You win if you can guess the 4 letter code in 12 turns. ' \
+          'After each guess, the mastermind will return the numbers 0 or 1. ' \
+          'Each "1" indicates that one of the guessed numbers is correct, but in the wrong place. ' \
+          'Each "2" indicates that one of the guessed numbers is correct and in the correct place. ' \
+          'You may guess any value between 2 - 7 inclusive. ' \
+          'Goodluck!!'
+  end
+
+  def update_game_count
+    @game_count.nil? ? @game_count = 1 : @game_count += 1
+  end
+
+  def check_guess_valid?(guess)
```

Comment:
Is there a reason you're passing in `guess`? It's an instance variable isn't it?

Reply:
yeah it is, i'll remove that param!